### PR TITLE
Status page

### PR DESCRIPTION
**Pull request by jonnydawg**
Status page now handles namespaces requests and will only show checks from the requested namespace.

 An empty namespace request will show checks from the `kube-system` and `kuberhealthy` namespaces. 

#280

### CODE REVIEW

File: cmd/kuberhealthy/kuberhealthy.go [hunk 1]
```diff
@@ -1010,8 +1010,12 @@ func (k *Kuberhealthy) healthCheckHandler(w http.ResponseWriter, r *http.Request
 		return err
 	}
 
+	// get URL query parameters if there are any
+	values := r.URL.Query()
+	namespace := values.Get("namespace")
```
Comment (joshulyne) [inline cmd/kuberhealthy/kuberhealthy.go]: what happens is there isn't any "namespace" value in the query?

Comment (joshulyne) [reply cmd/kuberhealthy/kuberhealthy.go]: ah just checked the `values.Get` command, it does return an empty string

Comment (joshulyne) [reply cmd/kuberhealthy/kuberhealthy.go]: looks good! 👍

**Comment by joshulyne** [comment]
lgtm!

**Comment by jonnydawg** [comment]
Where should these changes be documented?

File: cmd/kuberhealthy/reflector.go [hunk 2]
```diff
@@ -51,8 +51,8 @@ func (sr *StateReflector) Start() {
 	sr.reflector.Run(sr.reflectorSigChan)
 }
 
-// CurrentStatuses returns the current summary of all checks as known by the cache
-func (sr *StateReflector) CurrentStatus() health.State {
+// CurrentStatus returns the current summary of all checks as known by the cache
```
Comment (integrii) [inline cmd/kuberhealthy/reflector.go]: What happens if I want to pull the state for all checks?  Do I use a blank input parameter?  How do I know to do that?

Add some details to the comment here about how you can use this to poll everything.

File: cmd/kuberhealthy/reflector.go [hunk 3]
```diff
@@ -74,6 +74,22 @@ func (sr *StateReflector) CurrentStatus() health.State {
...
 
+		// if there is a requested namespace, then filter out checks from other namespaces
+		if len(namespace) != 0 {
+			if khState.GetNamespace() != namespace {
+				log.Debugln("Skipping", khState.GetName(), "because it is not from the", namespace, "namespace")
+				continue
+			}
+		}
+
+		// if there is no requested namespace, show checks from "kube-system" and "kuberhealthy"
```
Comment (integrii) [inline cmd/kuberhealthy/reflector.go]: this is a preference to us as the project owners, but is an unlikely assumption for all other users.  I would make the default "blank" functionality pick everything.

**Pull request merged**